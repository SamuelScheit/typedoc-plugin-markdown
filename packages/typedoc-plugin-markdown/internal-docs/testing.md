---
title: Testing guide
description: Guidance of running tests and creating fixtures.
---

# Testing

## Running tests

```bash
npm run test
```

If you require to update the snapshots run the following command:

```bash
npm run test:update
```

## Test Fixtures

A series of snapshot tests are used to ensure the output of the plugin is consistent.

- Example code for fixtures are located at `/test/fixtures/src`.
- Fixtures output scenarios are setup using a config file (`/test/fixtures/config.ts`) that provides a mechanism to easily provide a combination of options.
