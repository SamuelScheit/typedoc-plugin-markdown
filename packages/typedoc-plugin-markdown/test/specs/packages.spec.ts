import { expectFileToEqual } from '@devtools/testing';

describe(`Packages`, () => {
  test(`should compile readmes for a packages`, () => {
    expectFileToEqual('packages', 'members', 'index.md');
    expectFileToEqual('packages', 'members', 'package-1/index.md', 1);
  });

  test(`should compile index page for packages`, () => {
    expectFileToEqual('packages', 'members', ['packages.md']);
    expectFileToEqual('packages', 'members', 'package-1/globals.md', 1);
    expectFileToEqual('packages', 'members', 'package-2/modules.md', 1);
  });

  test(`should compile member page for packages`, () => {
    expectFileToEqual('packages', 'members', [
      'package-1/interfaces/PackageInterface.md',
    ]);
  });

  test(`should compile relative entry module package`, () => {
    expectFileToEqual('packages', 'members', 'package-6.md', 1);
  });
  test(`should compile member for relative entry module package`, () => {
    expectFileToEqual(
      'packages',
      'members',
      'package-6/module1/interfaces/PackageInterface1.md',
      1,
    );
  });

  test(`should compile index for a single package`, () => {
    expectFileToEqual('package', 'members', 'README.md');
  });

  test(`should compile member page for a single package`, () => {
    expectFileToEqual('package', 'members', 'interfaces/PackageInterface.md');
  });
});
