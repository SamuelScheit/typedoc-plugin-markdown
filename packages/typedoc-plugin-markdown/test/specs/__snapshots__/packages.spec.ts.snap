// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Packages should compile index page for packages: (Output File Strategy "members") (Option Group "1") 1`] = `
"**packages-example** • [Readme](index.md) \\| [Packages](packages.md)

***

# packages-example

## Packages

- [package-1](package-1/index.md)
- [package-2](package-2/index.md)
"
`;

exports[`Packages should compile index page for packages: (Output File Strategy "members") (Option Group "1") 2`] = `
"[packages-example](../packages.md) \\> **package-1** • [Readme](index.md) \\| [Exports](exports.md)

***

[packages-example](../packages.md) / package-1

# package-1

## Interfaces

- [PackageInterface](interfaces/PackageInterface.md)
"
`;

exports[`Packages should compile index page for packages: (Output File Strategy "members") (Option Group "1") 3`] = `
"[packages-example](../packages.md) \\> **package-2** • [Modules](index.md)

***

[packages-example](../packages.md) / package-2

# package-2

## Modules

- [module-1](module-1/index.md)
- [module-2](module-2/index.md)
"
`;

exports[`Packages should compile member page for packages: (Output File Strategy "members") (Option Group "1") 1`] = `
"[packages-example](../../packages.md) \\> **package-1** • [Readme](../index.md) \\| [Exports](../exports.md)

***

[packages-example](../../packages.md) / [package-1](../exports.md) / PackageInterface

# Interface: PackageInterface
"
`;

exports[`Packages should compile readmes for a packages: (Output File Strategy "members") (Option Group "1") 1`] = `
"**packages-example** • [Readme](index.md) \\| [Packages](packages.md)

***

# MY README

> Some block quote

Some text.

## Readme heading

Some text.
"
`;

exports[`Packages should compile readmes for a packages: (Output File Strategy "members") (Option Group "1") 2`] = `
"[packages-example](../packages.md) \\> **package-1** • [Readme](index.md) \\| [Exports](exports.md)

***

# PACKAGE 1 README

Readme content
"
`;

exports[`Packages should compile readmes for a packages: (Output File Strategy "members") (Option Group "2") 1`] = `
"**packages-example** • [Packages](index.md)

***

# MY README

> Some block quote

Some text.

## Readme heading

Some text.

## Packages

- [package-1](package-1/index.md)
- [package-2](package-2/index.md)
"
`;
